Add support for OpenAI latest text embedding models

yabinmeng opened the pull request:
# Overview

OpenAI has rolled out 2 new text embedding modes with the capability of  having reduced output dimension size without impacting the accuracy
- text-embedding-3-large
- text-embedding-3-small

Update NW program to support these 2 models and the new feature

yabinmeng committed:
Add support for OpenAI latest text embedding models with reduced output dimension size

yabinmeng committed:
Minor update of default model prefix name

phact commented on README.md:
```diff
@@ -44,18 +44,20 @@ positional arguments:
   -m MODEL_NAME, --model_name MODEL_NAME
-                        model name to use for generating embeddings, i.e. ada-002, textembedding-gecko, or intfloat/e5-large-v2
+                        model name to use for generating embeddings, i.e. text-embedding-ada-002, textembedding-gecko, or intfloat/e5-large-v2
+  -d DIMENSION_SIZE, --dimension_size DIMENSION_SIZE
```
hm maybe we should call this reduced_dimension_size to avoid confusion?

yabinmeng replied:
Changed to `[-rd REDUCED_DIMENSION_SIZE]`

```
-rd REDUCED_DIMENSION_SIZE, --reduced_dimension_size REDUCED_DIMENSION_SIZE
                        Reduced (output) dimension size. Only supported in models (e.g. OpenAI text-embedding-3-xxx) that have this feature. Ignored otherwise!
```

yabinmeng committed:
Naming change to the "reduced dimension size"

yabinmeng committed:
1) Fix zero embedding detection log
2) Fix final indice and distance dataset having one extra clumn 'RowNum'
3) Update program execution output to make the execution progress clearer
4) Update unit test cases
